src/storage/data_table.cpp

```diff
@@ -122,23 +131,79 @@ bool DataTable::Update(transaction::TransactionContext *const txn, const TupleSl
   return true;
 }
 
+void DataTable::CheckMoveHead(std::list<RawBlock *>::iterator block) {
+  common::SpinLatch::ScopedSpinLatch guard_head(&header_latch_);
+  if (block == insertion_head_) {
+    // If the header block is full, move the header to point to the next block
+    insertion_head_++;
+  }
+
+  // If there are no more free blocks, create a new empty block and  point the insertion_head to it
```

Comment:
Drop this code.  `DataTable::Insert` already handles this case for you (which is necessary for correctness) and I don't think there is a compelling reason to do eager allocation of a 1 MiB chunk of memory.

Reply:
This function is called by the Insert function. I may misunderstand your comments. 'DataTable::Insert already handles this case for you' Insert handles the case by calling the checkMoveHead function.